src/main/java/com/gluonhq/substrate/target/AndroidTargetConfiguration.java

```diff
@@ -87,20 +87,20 @@
             "-landroid", "-llog", "-lffi", "-llibchelper");
     private final List<String> javafxLinkFlags = Arrays.asList("-Wl,--whole-archive",
             "-lprism_es2_monocle", "-lglass_monocle", "-ljavafx_font_freetype", "-ljavafx_iio", "-Wl,--no-whole-archive",
-            "-lGLESv2", "-lEGL", "-lfreetype");
+            "-lGLESv2", "-lEGL", "-lfreetype", "-lstdc++");
     private final String capLocation = ANDROID_NATIVE_FOLDER + "cap/";
 
     public AndroidTargetConfiguration(ProcessPaths paths, InternalProjectConfiguration configuration) throws IOException {
         super(paths,configuration);
 
         this.sdk = fileDeps.getAndroidSDKPath().toString();
         this.ndk = fileDeps.getAndroidNDKPath().toString();
-        this.hostPlatformFolder = configuration.getHostTriplet().getOsArch();
+        this.hostPlatformFolder = Paths.get(this.ndk, "toolchains", "llvm", "prebuilt", configuration.getHostTriplet().getOsArch()).toString();
 
-        Path ldguess = Paths.get(this.ndk, "toolchains", "llvm", "prebuilt", hostPlatformFolder, "bin", "ld.lld");
+        Path ldguess = Paths.get(hostPlatformFolder, "bin", "ld.lld");
         this.ldlld = Files.exists(ldguess) ? ldguess : null;
 
-        Path clangguess = Paths.get(this.ndk, "toolchains", "llvm", "prebuilt", hostPlatformFolder, "bin", "clang");
+        Path clangguess = Paths.get(hostPlatformFolder, "bin", "clang");
```

Comment:
`objdumpguess` has to be also refactored.

The error messages in `compile()` and `link()` (`if (ldlld == null) throw new IOException()...`) need that refactoring too.